webapp/sources/rudder/rudder-rest/src/test/scala/com/normation/rudder/rest/TraitTestApiFromYamlFiles.scala

```diff
@@ -290,72 +276,118 @@ trait TraitTestApiFromYamlFiles extends Specification with BoxSpecMatcher with J
+    ZIO.foreach(files) {
       case (name, yamls) =>
-        // "." are breaking description, we need to remove it from file name
-        s"For each test defined in '${name.split("\\.").head}'" should {
-          Fragment.foreach(yamls.map(readSpecification).zipWithIndex.toSeq) { x =>
-            x match {
-              case (eb: EmptyBox, i) =>
-                val f = eb ?~! s"I wasn't able to run the ${i}th tests in file ${name}"
-                ApplicationLogger.error(f.messageChain)
-
-                s"[${i}] failing test ${i}: can not read description" in {
-                  ko(s"The yaml description ${i} in ${name} cannot be read: ${f.messageChain}")
-                }
-
-              case (Full(test), i) =>
-                s"[${i}] ${test.description}" in {
-                  val mockReq = new MockHttpServletRequest("http://localhost:8080")
-                  mockReq.method = test.method
-                  val p       = test.url.split('?')
-                  mockReq.path = p.head // exists, split always returns at least one element
-                  mockReq.queryString = if (p.size > 1) p(1) else ""
-                  mockReq.body = test.queryBody.getBytes(StandardCharsets.UTF_8)
-                  mockReq.headers = test.headers.map { h =>
-                    val parts = h.split(":")
-                    (parts(0), List(parts.tail.mkString(":").trim))
-                  }.toMap
-                  // query string may have already set some params
-                  mockReq.parameters = mockReq.parameters ++ test.params
-                  mockReq.contentType = mockReq.headers.get("Content-Type").flatMap(_.headOption).getOrElse("text/plain")
-
-                  restTest.execRequestResponse(mockReq)(response => {
-                    val responseBox = response.map(cleanResponse(_))
-                    responseBox.map(_._1) must equalsBoxStrict(test.responseCode)("response code") and (
-                      responseBox.map(_._2) must equalsBoxJson(test.responseContent)("response content")
-                    )
-                  })
-                }
-            }
+        // we can't have "." in test description, it breaks junit test runner in intellij
+        (suite(s"Tests defined in '${name.split("\\.").head}' yaml") {
+
+          val tests: ZIO[Any, Throwable, List[Spec[Any, Nothing]]] = {
+            (ZIO.foreach(yamls.map(readSpecification).zipWithIndex) { x =>
+              (x match {
+                case (eb: EmptyBox, i) =>
+                  val f = eb ?~! s"I wasn't able to run the ${i}th tests in file ${name}"
+                  ApplicationLogger.error(f.messageChain)
+
+                  // erg. There must be a better way
+                  zio.test.test(
+                    s"[${i}] failing test ${i}: can not read description. The yaml description ${i} in ${name} cannot be read: ${f.messageChain}"
+                  )(
+                    assert(true)(Assertion.isFalse)
+                  )
```

Comment:
I wasn't able to find a `test.failWithMessage(....)` method. I'm sure it exists somehow, but in the meantime, that will do.